### PR TITLE
Add `BTCardClientErrorTypeCardAlreadyExists`

### PR DESCRIPTION
**Pull request by sarahkoop**
### Summary of changes

- Add explicit error case for error code returned from Gateway when a card already exists in the vault.

### Checklist

- [x] Added a changelog entry

- @jaxdesmarais 
- @sarahkoop

### CODE REVIEW

File: Sources/BraintreeCard/Public/BraintreeCard/BTCardClient.h [hunk 1]
```diff
@@ -26,6 +26,9 @@ typedef NS_ENUM(NSInteger, BTCardClientErrorType) {
 
     /// Customer provided invalid input
     BTCardClientErrorTypeCustomerInputInvalid,
+    
+    /// Card already exists as a saved payment method
+    BTCardClientErrorTypeCardAlreadyExists,
```
Comment (sarahkoop) [inline Sources/BraintreeCard/Public/BraintreeCard/BTCardClient.h:31]: Pending discussion of whether or not this should be considered a breaking change

File: Sources/BraintreeCard/BTCardClient.m [hunk 2]
```diff
@@ -237,4 +233,24 @@ - (BOOL)isGraphQLEnabledForCardTokenization:(BTConfiguration *)configuration {
     return graphQLFeatures && [graphQLFeatures containsObject:BTCardClientGraphQLTokenizeFeature];
 }
 
+- (NSError *)cardTokenizationErrorForError:(NSError *)error {
```
Comment (crookedneighbor) [inline Sources/BraintreeCard/BTCardClient.m:236]: I find this name a little confusing. Can you explain the `ErrorForError` bit?

File: Sources/BraintreeCard/Public/BraintreeCard/BTCardClient.h (hunk 1 above)
Comment (crookedneighbor) [reply Sources/BraintreeCard/Public/BraintreeCard/BTCardClient.h:31]: I don't think this is a breaking change. As long as the merchant is handling a catch all error of "something went wrong" or just passing back the error message from the error directly, it should be fine.

File: Sources/BraintreeCard/BTCardClient.m (hunk 2 above)
Comment (sarahkoop) [reply Sources/BraintreeCard/BTCardClient.m]: We updated this helper method name to clarify. Let us know if it needs more refining 👍

File: Sources/BraintreeCard/BTCardClient.m [hunk 3]
```diff
@@ -237,4 +244,20 @@ - (BOOL)isGraphQLEnabledForCardTokenization:(BTConfiguration *)configuration {
     return graphQLFeatures && [graphQLFeatures containsObject:BTCardClientGraphQLTokenizeFeature];
 }
 
+- (NSError *)constructCallbackErrorResponseForErrorCode:(BTJSON *)errorCode error:(NSError *)error {
```
Comment (sshropshire) [inline Sources/BraintreeCard/BTCardClient.m:247]:
```suggestion
- (NSError *)constructCallbackErrorForErrorCode:(String *)errorCode error:(NSError *)error {
```
Super nit.

**Review by sshropshire — APPROVED** [review]

**Review by cgdibble — APPROVED** [review]

**Pull request merged**